detect.go

```diff
@@ -54,21 +58,36 @@ func Detect() packit.DetectFunc {
 			}
 		}
 
-		return packit.DetectResult{
+		npmDependency := packit.BuildPlanRequirement{
+			Name: Npm,
+			Metadata: BuildPlanMetadata{
+				Build: true,
+			},
+		}
+
+		cPythonDependency := packit.BuildPlanRequirement{
+			Name: Cpython,
+			Metadata: BuildPlanMetadata{
+				Build: true,
+			},
+		}
+
+		result := packit.DetectResult{
 			Plan: packit.BuildPlan{
 				Provides: []packit.BuildPlanProvision{
 					{Name: NodeModules},
 				},
 				Requires: []packit.BuildPlanRequirement{
 					nodeDependency,
-					{
-						Name: Npm,
-						Metadata: BuildPlanMetadata{
-							Build: true,
-						},
-					},
+					npmDependency,
 				},
 			},
-		}, nil
+		}
+
+		if pythonNeeded {
```

Comment:
We are not going to want to add the cpython layer for ubi, instead we would have the extension install it.  This will likely require changes to the cpython buildpack to not install if the requirement is already statisfied when the buildpack runs.  

Since that will take landing changes in the cpython library and we know that all of the build images used with Node.js applications and ubi currently include python, what I suggest is an if which wraps this and does not add the requirement if the stack is ubi based.

Reply:
I'll also add that in addition to python most native addons will require the full C/C++ compiler chain. Unless that is on the lighter weight stack already many if not most addons won't compile even if python is present.

As one of the examples listed in the original issue requesting the feature, it includes C/C++  - https://github.com/mscdex/cpu-features/blob/master/src/binding.cc 

So as a sanity check, does the base stack that they wanted to use include the C/C++ compiler but not python? Otherwise simply adding python won't be enough.

Reply:
> Since that will take landing changes in the cpython library and we know that all of the build images used with Node.js applications and ubi currently include python, what I suggest is an if which wraps this and does not add the requirement if the stack is ubi based.

But isn't that a generic "problem" with cnb buildpacks? Dependencies are always installed, regardless if they are present on the `stack` or not. Some for `node` or `jdk` or am I missing something?

> what I suggest is an if which wraps this and does not add the requirement if the stack is ubi based.

I thought the `STACK_ID` is deprecated.

Reply:
> Otherwise simply adding python won't be enough.

Good point, but strangely it was enough in the cases I tested. The stack I test with has no compiler chain. I will check.

Reply:
> So as a sanity check, does the base stack that they wanted to use include the C/C++ compiler but not python? Otherwise simply adding python won't be enough.

You are right. This only helps with stack that have a compiler, but no `python` (e.g. `paketo-base`). I am not so sure anymore if this has enough benefit or if you should just use a larger stack (e.g. `paketo-full`) and simply the smaller run-image.